File: python/invariance/resources/receipts.py
from __future__ import annotations

from typing import Any

from ..http_client import HttpClient
from ..types import Receipt, ReceiptQuery


class ReceiptsResource:
    def __init__(self, http: HttpClient) -> None:
        self._http = http

    async def submit(self, receipts: list[Receipt]) -> dict[str, int]:
        return await self._http.post("/v1/receipts", {"receipts": receipts})

    async def query(self, opts: ReceiptQuery | None = None) -> list[Receipt]:
        return await self._http.get("/v1/receipts", params=opts)

File: packages/typescript/src/resources/receipts.ts
import type { HttpClient } from '../http.js';
import type { Receipt, ReceiptQuery } from '../types/receipt.js';

export class ReceiptsResource {
  constructor(private http: HttpClient) {}

  async submit(receipts: Receipt[]): Promise<{ inserted: number }> {
    return this.http.post<{ inserted: number }>('/v1/receipts', { receipts });
  }

  async query(opts?: ReceiptQuery): Promise<Receipt[]> {
    return this.http.get<Receipt[]>('/v1/receipts', {
      params: opts as Record<string, string | number | undefined>,
    });
  }
}
